io/include/pcl/io/hdl_grabber.h

```diff
@@ -183,12 +206,17 @@ namespace pcl
       float
       getMaximumDistanceThreshold ();
 
+      /** \brief Returns the maximum number of lasers
+      */
+      static uint8_t
+      getMaximumNumberOfLasers ();
```

Comment:
This method is shadowed in the deriving class. This is a recipe for troubles!
Suppose you have `HDLGrabber* grabber = new VLPGrabber;`.
What do you think `grabber->getMaximumNumberOfLasers()` will return?

Please change into virtual const.

Reply:
> Suppose you have `HDLGrabber* grabber = new VLPGrabber;`

I argue that use case is not really applicable given the current inheritance scheme (poorly implemented).

Reply:
Well, I did not invent this, this is exactly use-case Tsukasa proposed: https://github.com/PointCloudLibrary/pcl/pull/1738#issuecomment-346638594

Reply:
Does your proposal mean this?

```patch
- static uint8_t
- getMaximumNumberOfLasers ();
+ virtual uint8_t
+ getMaximumNumberOfLasers () const;

  uint8_t
- pcl::HDLGrabber::getMaximumNumberOfLasers ()
+ pcl::HDLGrabber::getMaximumNumberOfLasers () const
  {
    return (HDL_MAX_NUM_LASERS);
  }
```
```patch
- static uint8_t
- getMaximumNumberOfLasers ();
+ virtual uint8_t
+ getMaximumNumberOfLasers () const;

  uint8_t
- pcl::VLPGrabber::getMaximumNumberOfLasers ()
+ pcl::VLPGrabber::getMaximumNumberOfLasers () const
  {
    return (VLP_MAX_NUM_LASERS);
  }
```

Reply:
Yes

Reply:
Fixed.